src/internal/types.ts

```diff
@@ -96,15 +97,25 @@ type IsBoundedString<T> = T extends string
  *
  * @see EnumerableStringKeyedValueOf
  */
-export type EnumerableStringKeyOf<T> = `${Exclude<keyof T, symbol>}`;
+export type EnumerableStringKeyOf<T> =
+  T extends Record<infer K, unknown> ? `${Exclude<K, symbol>}` : never;
+
+/**
+ * Extracts the value type from an object type T.
+ */
+export type InferredRecordValue<T> = T extends EmptyObject
```

Comment:
The "Inferred" part in the name does not describe what this type is; it's how it's calculated. This doesn't communicate clearly what this type is. Similarly, Records, conceptually, are a subset of all objects for simple mappings. It's true that when doing complex typing, it's a nice shorthand for more complicated syntax, but it's not the intent of the term.

Naming stuff is often a matter of taste, so I'm not putting my foot down on this, but having read a lot of old code to try and detangle Remeda in the first place, a good name goes a long way to simplify long-term maintenance.

It seems like the correct name here would be `Value` or `ValuesOf`, as in `ValuesOf<MyType>`

Reply:
Thank you for your suggestion, `ValuesOf` is indeed a more precise name.